### ISSUE TITLE
[Invisible mode] remote changes are not synced correctly which causes conflictions like `\end{document}nt}ent}nt}`

### ISSUE THREAD

**Issue by e-zz** [iamhyc/Overleaf-Workshop#109]
### Describe the bug
Remote changes are automatically pulled which is super convenient. 
But sometimes the changes are not correctly merged into local files (even in the case without any confliction)

For example, in a project I created a tex file 
``` tex
\documentclass{article}
\begin{document}
(Type yourhere.)
\end{document}
```
At the same time I created a local repliica for it and then turned on invisible mode.

### Modify the file on overleaf
On Overleaf, I did some random input test. In the end the file was modified as 
``` tex
\documentclass{article}
\begin{document}
(Type yourhere.)
$\cos x$

1
2
3
4
5
6
7
8
9
d d 
j 

d 
d d

j w
d d

jww d
j d

\end{document}
```

### changes are synced wiith the local version, but not in an expected way
At the same time, the plugin pulls the changes from overleaf. But those lines I marked (by `<=`) are a bit off
``` tex
\documentclass{article}
\begin{document}
(Type yourhere.)
$\cos x$

1
2
3
4
5
6
7
8
9
d d  d   <=
j 

d 
d d

j w
d d

jww d
j d

\end{document}nt}ent}nt}  <= 
```

I tested a few rounds and it's quite easy to reproduce the bug, especially the pattern `\end{document}nt}ent}nt}`.
Notice that no we haven't made any change locally. So it's purely a problem of pulling the changes from overleaf.

### Expected behavior
Remote changes are correctly synced to local files in the invisible mode.

### How To Reproduce
- create a file (as the one above)
- Enter invisible mode
- keep editing it in Overleaf (random input test)

Just within few lines the issue can be noticed in the local version.

### Environment
 - **Overleaf Workshop Extension version**: 0.9.2
 - **VS Code version**: 1.85.1
 - **Overleaf Edition**: Official 

### [Optional] Developer Logs
> If applicable, please paste the logs inside the details section below.
> Please find the logs via: Title bar **"Help"** > **"Toggle Developer Tools"** > **"Console"**.

<details>
  ...
</details>

**Comment by iamhyc** [comment]
Hi @e-zz , the latest temporary extension may have solved the problem. We have verified it works normally in common cases.

Please also help check : https://github.com/iamhyc/Overleaf-Workshop/actions/runs/8122179672

**Comment by iamhyc** [comment]
This issue is closed with the PR #113 merged.

If there is still any remaining problem, please feel free to reopen it or create new issue.

If there is nothing wrong, I will include the changes in a recent release.

**Comment by e-zz** [comment]
> Hi @e-zz , the latest temporary extension may have solved the problem. We have verified it works normally in common cases.
> 
> Please also help check : https://github.com/iamhyc/Overleaf-Workshop/actions/runs/8122179672

Hi
Now it works like a charm. Also I noticed that changes are synced much faster now. Bravo!
